pkg/connection/credentials.go

```diff
@@ -6,17 +6,18 @@ type Credentials interface {
 	// Returns true if we can authenticate at all, false otherwise.
 	HasAuthentication() bool
 
-	// Returns the authentication triplet in this order: username, password and
-	// system token; or an error if the implementing structure cannot return
-	// this triplet for whatever reason.
-	Triplet() (string, string, string, error)
+	// Token returns the current system used to detect duplicated systems. This
+	// token gets rotated on each non read operation.
+	Token() (string, error)
```

Comment:
I split the `Update` and `Triplet` methods into `Login` and `Token` based methods since both are used in different places and I did not want to query the `Login` when I actually only interested in the `Token`.

no need for this construction this way:

```golang
login, pw, _, err := conn.Credentials.Triplet()
...
token := response.Header.Get('System-Token')
...
conn.Credentials.Update(login, pw, token)
```